examples/some-verifier/src/db.rs

```diff
@@ -0,0 +1,299 @@
+        .filter_map(|(name, val)| val.map(|v| (name, v)))
+    }
+}
+
+pub struct PlatformEntry {
+    pub id: String,
+    pub cred_id: CredentialHolderId,
+    pub username: String,
+}
+
+impl PlatformEntry {
+    fn columns<'a>(
+        &'a self,
+        verification_id: &'a i64,
+    ) -> impl Iterator<Item = (&'static str, &(dyn ToSql + Sync))> {
+        [
+            (ID_COLUMN, &self.id as &(dyn ToSql + Sync)),
+            (
+                CRED_ID_COLUMN,
+                self.cred_id.public_key.as_bytes() as &(dyn ToSql + Sync),
+            ),
+            (
+                VERIFICATION_ID_COLUMN,
+                verification_id as &(dyn ToSql + Sync),
+            ),
+            (USERNAME_COLUMN, &self.username),
+        ]
+        .into_iter()
+    }
+}
+
+pub type DbResult<T> = Result<T, tokio_postgres::Error>;
+
+impl Database {
+    pub async fn connect(db_config: tokio_postgres::Config) -> DbResult<Self> {
+        let (client, connection) = db_config.connect(NoTls).await?;
+
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                tracing::error!("connection error: {}", e);
+            }
+        });
+
+        Ok(Self {
+            client: RwLock::new(client),
+        })
+    }
+
+    /// Returns the verification for a given social media account if it exists.
+    pub async fn get_verification(
+        &self,
+        id: &str,
+        platform: Platform,
+    ) -> DbResult<Option<DbVerification>> {
+        // The base statement
+        let mut statement =
+            format!("SELECT {PRESENTATION_COLUMN}, {FIRST_NAME_COLUMN}, {LAST_NAME_COLUMN}");
+
+        // Additional columns to select and joins to perform built from the supported platforms.
+        let (columns, joins) = Platform::SUPPORTED_PLATFORMS
+            .into_iter()
+            .map(|platform| {
+                let column = format!(
+                    "{0}.{USERNAME_COLUMN} AS {1}, {}.{ID_COLUMN} AS {2}",
+                    platform.table_name(),
+                    platform.username_alias(),
+                    platform.id_alias()
+                );
+                let join = format!(
+                    "JOIN {0} ON {0}.{VERIFICATION_ID_COLUMN}={VERIFICATIONS_TABLE}.{ID_COLUMN}",
+                    platform.table_name(),
+                );
+                (column, join)
+            })
+            .fold(
+                (String::new(), String::new()),
+                |(mut columns, mut joins), (column, join)| {
+                    columns.push_str(&format!(", {}", column));
+                    joins.push_str(&format!(" {}", join));
+                    (columns, joins)
+                },
+            );
+
+        statement.push_str(&columns);
+        statement.push_str(&format!(" FROM {VERIFICATIONS_TABLE}"));
+        statement.push_str(&joins);
+        statement.push_str(&format!(
+            " WHERE {}.{ID_COLUMN} = $1",
+            platform.table_name()
+        ));
```

Comment:
Referring to a discussion in office regarding sql injection. Is it bad to build statements like this (besides being a bit hard to follow what the final statement will be)?

What we try to do here is to make something which works as `Platform::SUPPORTED_PLATFORMS` is extended with more members.

Reply:
@abizjak

Reply:
I revised this quite a bit. Now the statements are simpler and constructed in a more direct way.